File: notesWep/backend/app/models/note.rb
class Note
  
  include Mongoid::Document
  include Mongoid::Timestamps

  # Fields
  field :user, type: BSON::ObjectId,  default: ''
  field :title, type: String, default: ''
  field :content, type: String,  default: ''
  field :bgColor, type: String,  default: ''
  field :imageURL, type: String, default: ''
  field :isShared, type: Mongoid::Boolean,  default: ''

  # Validations
  validates_presence_of :_id, :title, :content
  validates_uniqueness_of :_id
end

File: notesWep/frontend/src/app/classes/note.ts
import ObjectId from 'bson-objectid';
// This class is used to model and handle individual notes created by users.
export class Note {
    _id?: ObjectId;
    title: string;
    content: string;
    user: ObjectId;
    bgColor: string;
    isShared: boolean;
    imageURL?: string;
    updated_at?: Date;
    created_at?: Date;
}